src/adaptation/composite.jl

```diff
@@ -48,12 +46,43 @@ function Base.show(io::IO, state::StanHMCAdaptorState)
     print(io, "window($(state.window_start), $(state.window_end)), window_splits(" * string(join(state.window_splits, ", ")) * ")")
 end
 
-################
-### Adaptors ###
-################
+### Composite adaptors
+
+abstract type CompositeAdaptor <: AbstractAdaptor end
+
+getM⁻¹(ca::CompositeAdaptor) = getM⁻¹(ca.pc)
```

Comment:
@yebai This is the only intention to introduce `CompositeAdaptor`. These functions was definted for `StanHMCAdaptor` and `NaiveHMCAdaptor` seperately. Please let me know if you think it's fine to introduce this in this PR. I'm fine to revert it as well if you think it's a bit of an over-kill for now.

Reply:
I find it a bit overkill and makes things slightly harder to follow here. Maybe revert it.

Reply:
No problem for me.